python/dgl/graphbolt/base.py

```diff
@@ -38,12 +38,30 @@
     "CSCFormatBase",
     "seed",
     "seed_type_str_to_ntypes",
+    "get_host_to_device_uva_stream",
+    "get_device_to_host_uva_stream",
 ]
 
 CANONICAL_ETYPE_DELIMITER = ":"
 ORIGINAL_EDGE_ID = "_ORIGINAL_EDGE_ID"
 
 
+# There needs to be a single instance of the uva_stream, if it is created
+# multiple times, it leads to multiple CUDA memory pools and memory leaks.
+def get_host_to_device_uva_stream():
+    """The host to device copy stream to be used for pipeline parallelism."""
+    if not hasattr(get_host_to_device_uva_stream, "stream"):
+        get_host_to_device_uva_stream.stream = torch.cuda.Stream(priority=-1)
+    return get_host_to_device_uva_stream.stream
```

Comment:
I know python is a very flexible language, but this is probably less error prone.

```
def get_singleton_host_to_device_uva_stream():
    _stream = None  # Private variable within the static method

    def get_stream():
        nonlocal _stream
        if _stream is None:
            _stream = torch.cuda.Stream(priority=-1)
        return _stream

    return get_stream()
```